### PR TITLE
Proxy from env

### PR DESCRIPTION
**Pull request by erikerikson**
This PR builds on top of https://github.com/serverless/serverless/pull/536 and would be great to get into v0.4.

To see just the changes added by this PR relative to 536, see https://github.com/serverless/serverless/commit/2d90a3f567e429e8118627f431ccf94ac440af98 .

Those changes are:
1. check the environment for relatively standard environment variables used to declare proxy use.
2. if provided, use that configuration to configure the use of a proxy in the aws-sdk.

Open question:
1. Are there any other libraries that require the use of a proxy?  If so, they will need to be configured as well.
2. Is this bug free?  The current state of master bars me from testing these changes but they worked in previous tests.

### CODE REVIEW

File: bin/serverless [hunk 1]
```diff
@@ -2,7 +2,18 @@
 
 'use strict';
 
-let argv = require('minimist')(process.argv.slice(2));
+let argv = require('minimist')(process.argv.slice(2)),
```
Comment (ac360) [inline bin/serverless]: @erikerikson Is there any way we can move this logic into `Serverless.js`?  We're trying to keep this file lean.

**Comment by ac360** [comment]
@erikerikson Can you submit your PRs to the `v0.5` branch so I can merge them in?

**Comment by erikerikson** [comment]
Absolutely.  Busy today but will get that in by tomorrow mid-day if not sooner

**Comment by erikerikson** [comment]
do you have an opinion about the proxy and session token changes being in separate PRs?

**Comment by erikerikson** [comment]
moved to https://github.com/serverless/serverless/pull/627 to target the v0.5 branch
closing...

**Pull request closed without merging**